### ISSUE TITLE
Add a `--test-list` flag to `zig test`

### ISSUE THREAD

**Issue by torshepherd** [ziglang/zig#15558]
# Background

Many test frameworks offer CLI flags to list tests that the test command would pick up without running them

- [GTest](https://google.github.io/googletest/advanced.html#listing-test-names) (C++)
- [Catch2](https://github.com/catchorg/Catch2/blob/devel/docs/command-line.md#listing-available-tests-tags-or-reporters) (C++)
- [Pytest](https://www.wireshark.org/docs/wsdg_html_chunked/ChTestsRunPytest.html) (Python)
- [Jest](https://jestjs.io/docs/cli#--listtests) (JS)

# Motivation

One benefit of adding this CLI option is enabling editor test integration.

[The Jest extension for vscode](https://marketplace.visualstudio.com/items?itemName=Orta.vscode-jest), for example, contributes discovered test cases to the [VSCode Test Explorer](https://code.visualstudio.com/api/extension-guides/testing). Adding this CLI option to `zig` would enable similar test integration in [vscode-zig](https://github.com/ziglang/vscode-zig), enabling auto-running tests on modification, inline test case failure reporting, and more.

If accepted, I would be happy to submit a PR that just adds this CLI option.

**Comment by matu3ba** [comment]
`zig test --test-list` feels redundant and for consistency `zig test -l` and/or `zig test --list` should be used similar to how  `zig build -l` lists the available top level steps.

**Comment by torshepherd** [comment]
Agreed. I just named it as such to be in-line with the other tes-specific CLI args

**Comment by torshepherd** [comment]
Can I start working on a PR that adds this?
